### PR TITLE
Use existing CRIU packages for testing instead of building from git

### PR DESCRIPTION
**Pull request by adrianreber**
This reverts commit dd81a8fe7dcc532f78d03291223806a877f820c7.

Now that the CRIU socket has the correct SELinux label checkpointing and
restoring works again on F28 and F29.

Related to #1962 #1891 #1890

### CODE REVIEW

**Comment by mheon** [comment]
Seeing checkpoint failures on F29 - are we expecting it to pass there?

**Comment by adrianreber** [comment]
I was expecting it to pass after it passed on my F28 and F29 test VMs. I need to look closer.

**Comment by adrianreber** [comment]
Closing this, as it is not yet ready.

**Comment by adrianreber** [comment]
Reopening for further testing.

File: test/e2e/checkpoint_test.go [hunk 1]
```diff
@@ -63,6 +60,13 @@ var _ = Describe("Podman checkpoint", func() {
 		result := podmanTest.Podman([]string{"container", "checkpoint", cid})
 		result.WaitWithDefaultTimeout()
 
+		cmd := exec.Command("cat", podmanTest.CrioRoot+"/vfs-containers/"+cid+"/userdata/dump.log")
+		out, _ := cmd.CombinedOutput()
+		fmt.Printf("combined out:\n%s\n", string(out))
+		cmd = exec.Command("ausearch", "-m", "AVC", "-m", "USER_AVC", "-ts", "recent")
+		out, _ = cmd.CombinedOutput()
+		fmt.Printf("combined out:\n%s\n", string(out))
+
```
Comment (cevich) [inline test/e2e/checkpoint_test.go]: Please feel free to do whatever debugging you need to in the CI scripts as well.  For example, ``set +e`` before the test runs/fails, then ``journalctl -lb 0`` to splat the logs.

**Comment by adrianreber** [comment]
With the latest debug output added I can now see that CRIU fails with:
```
(00.014936) Error (criu/lsm.c:94): Non unconfined selinux contexts not supported system_u
(00.014944) ----------------------------------------
(00.014948) Error (criu/cr-dump.c:1386): Dump core (pid: 26575) failed with -1
```

Which leads to this in the CRIU code base:
```
        /*
         * Since SELinux attributes can be finer grained than at the task
         * level, and we currently don't try to dump any of these other bits,
         * let's only allow unconfined profiles, which look something like:
         *
         *      unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
         */
```

Which means there is probably CRIU support missing. Next step is to get better debug output from CRIU to understand where it finds the attribute it errors out on.

I will rerun the test cases with an instrumented version of CRIU.

**Comment by rhatdan** [comment]
@adrianreber Please rebase.

**Comment by adrianreber** [comment]
@cevich I updated `contrib/cirrus/lib.sh` to use existing packages whenever possible, instead of building from git. I found out that the distribution provided packages on Fedora are working correctly compared to the version from git with which we overwrote the distribution package. The CRIU version from git was necessary in the beginning, when the distribution packages were still missing features.

I tried to put all the logic in `install_criu`, but I was not really able to test it. Do you have a way to see if my changes to `contrib/cirrus/lib.sh` are actually working? Only the commit `Use existing CRIU packages` is important.

**Comment by cevich** [comment]
@adrianreber the tests that run on the PR use whatever code is there (including the CI scripts), so that's the easiest way to test changes.  Sounds like #1936 will do some fixing in this area as well, I'll take a peek whatcha got here...

File: contrib/cirrus/lib.sh [hunk 2]
```diff
@@ -248,18 +248,46 @@ install_conmon(){
...
+name=Copr repo for criu-el7 owned by adrian
+baseurl=https://copr-be.cloud.fedoraproject.org/results/adrian/criu-el7/epel-7-$basearch/
+type=rpm-md
+skip_if_unavailable=True
+gpgcheck=1
+gpgkey=https://copr-be.cloud.fedoraproject.org/results/adrian/criu-el7/pubkey.gpg
+repo_gpgcheck=0
+enabled=1
+enabled_metadata=1
+EOF
```
Comment (cevich) [inline contrib/cirrus/lib.sh]: No tests are happening with CentOS/RHEL, hopefully there will be someday soon.  Until then, this is plenty helpful when it comes time to get them running.

**Comment by cevich** [comment]
...looked, yep that's cool @adrianreber and It's going to conflict with my PR but I can deal with that NP.   I think the changes you made are plenty good'nuff and the tests pass, so :+1: 

/lgtm

**Comment by adrianreber** [comment]
I removed my debug test commit and now this PR only includes the changes to `contrib/cirrus/lib.sh`. CI fails, however, with something unrelated to my change. `gofmt` complains about code which more than one year old. Strange.

**Comment by rhatdan** [comment]
There is a major difference between gofmt 1.10 and gofmt 1.11  You are probably hitting this issue.

**Comment by cevich** [comment]
> gofmt complains about code which more than one year old. Strange.

Yeah, def. not you, it's affecting other PRs and testing on master as well.  Fix is in the works.

**Comment by adrianreber** [comment]
I renamed this PR as it now only changes the CI setup to use existing CRIU packages instead of building CRIU from git. Once this is merged and new CI VM exist the next step would be to re-enable CRIU testing.

**Comment by adrianreber** [comment]
@cevich: will my changes in this PR be immediately part of the CI VMs or do they need to be re-created?

**Comment by cevich** [comment]
> do they need to be re-created?

Oh right, thanks for the reminder.  They way it's currently written, the images need to be re-built.  This changes with #1936 where we'll do these things at test-time.  That PR also adds more automation around this step.  Thanks for the reminder, I'll take care of that.

**Comment by cevich** [comment]
@adrianreber oh one more thing :smile:  For ubuntu, please connect with @lsm5 about getting packages built.  He's got some automation around something similar to copr (but for ubuntu).  We'll need this before too long, since #1936 moves us off of source-installs on all platforms.  I'm focusing on getting Fedora tests working first, then will work on Ubuntu, so having a package isn't super-duper urgent, just mildly :smile:

**Comment by cevich** [comment]
Oh, my bad, I see you've already got ``criu/ppa``, okay, NP then that works too.

**Comment by adrianreber** [comment]
@cevick, good that that PPA works. I am publishing latest CRIU packages as part of my upstream involvement in CRIU for Ubuntu (PPA) and CentOS (COPR).

**Comment by cevich** [comment]
Images updated for general use in #2042 and I migrated your work here into #1936.  Thanks again!

**Pull request merged**